src/arithmetics/mod.rs

```diff
@@ -689,3 +689,141 @@ pub fn max_usize_arr<C: Config>(
 
     Usize::from(max_var)
 }
+
+pub fn extrapolate_uni_poly_deg_1<C: Config>(builder: &mut Builder<C>, p_i: &Array<C, Ext<C::F, C::EF>>, eval_at: Ext<C::F, C::EF>) -> Ext<C::F, C::EF> {
+    builder.assert_usize_eq(p_i.len(), Usize::from(2));
+
+    let x0: Ext<C::F, C::EF> = builder.constant(C::EF::ZERO);
+    let x1: Ext<C::F, C::EF> = builder.constant(C::EF::ONE);
```

Comment:
As `builder.constant(C::EF::ZERO)` takes 4 cycles, we should cache them inside `once_cell`.